jax_unirep/evotuning.py

```diff
@@ -196,6 +195,126 @@ def step(i, state):
     return get_params(state)
 
 
+def avg_loss(sequences, params):
+    """
+    Return average loss of a set of parameters,
+    on a set of sequences.
+
+    :param sequences: sequences (in standard AA format)
+    :param params: parameters (i.e. from training)
+    """
+    xs, ys = length_batch_input_outputs(sequences)
+
+    sum_loss = 0
+    for x, y in zip(xs, ys):
+        sum_loss += evotune_loss(params, inputs=x, targets=y) * len(x)
+
+    return sum_loss / len(sequences)
+
+
+def fit_manual(
```

Comment:
Some issues I'm seeing, now that I see the extra commits coming in. Firstly, it seems to me that there's `fit_manual` has a lot copied from `fit`, but tweaked slightly. This isn't the best coding practice for a software library. 

Secondly, the API has extra pieces compared to `fit`, making the API inconsistent. In particular, the `in_val_seqs`, `out_val_seqs`, `proj_name`, and `steps_per_print` kwargs don't have semantic counterparts in `evotune()`.

Finally, `fit_manual` doesn't do one "unit" thing - it fits, and also handles data I/O (to disk). Generally, I have seen libraries leave data I/O to a separate function that the user has to explicitly call, rather than embed it inside a function. In a "library" function, it's preferable for functions to be "unit".